lib/blocks.php

```diff
@@ -181,6 +181,8 @@ function gutenberg_register_core_block_styles( $block_name ) {
 
 		// Add a reference to the stylesheet's path to allow calculations for inlining styles in `wp_head`.
 		wp_style_add_data( "wp-block-{$block_name}", 'path', gutenberg_dir_path() . $style_path );
+	} else {
+		wp_register_style( "wp-block-{$block_name}", false );
```

Comment:
What if WordPress core registers the same style handle?

Reply:
If core registers the same handle it will be for the block's styles - which would mean the block has a `style.css` file in which case we're still OK (this won't override existing registered styles, it will only add it if not already registered).
I did check however, and core doesn't register anything that would conflict

Reply:
At some point, we will have the same style handles registered in the core, so we need somehow to account for it.

Reply:
https://github.com/WordPress/wordpress-develop/blob/2382765afa36e10bf3c74420024ad4e85763a47c/src/wp-includes/class.wp-dependencies.php#L236-L238

OK, I see what you mean. If the style is already registered then it will do nothing here. The only case when it would be unexpected is when WordPress core would define some styles, but the plugin would prefer to remove them which is very unlikely.

Reply:
Yes! That's exactly what I mean :+1: